Kernel/Firmware/EFI/Protocols/MediaAccess.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2024, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/EnumBits.h>
+#include <Kernel/Firmware/EFI/EFI.h>
+#include <Kernel/Firmware/EFI/SystemTable.h>
+
+// https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html
+
+namespace Kernel::EFI {
+
+// See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#efi-file-protocol-open
+enum class FileOpenMode : u64 {
+    Read = 0x1,
+    Write = 0x2,
+    Create = 0x8000000000000000,
+};
+AK_ENUM_BITWISE_OPERATORS(FileOpenMode)
+
+// See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#efi-file-protocol-open
+enum class FileAttribute : u64 {
+    None = 0x0,
+    ReadOnly = 0x1,
+    Hidden = 0x2,
+    System = 0x4,
+    Reserved = 0x8,
+    Directory = 0x10,
+    Archive = 0x20,
+};
+AK_ENUM_BITWISE_OPERATORS(FileAttribute)
+
+// EFI_FILE_INFO: https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#efi-file-info
+struct FileInfo {
+    static constexpr GUID guid = { 0x09576e92, 0x6d3f, 0x11d2, { 0x8e, 0x39, 0x00, 0xa0, 0xc9, 0x69, 0x72, 0x3b } };
+
+    u64 size;
+    u64 file_size;
+    u64 physical_size;
+    Time create_time;
+    Time last_access_time;
+    Time modification_time;
+    FileAttribute attribute;
+    char16_t file_name[];
+};
+
+// EFI_FILE_IO_TOKEN: See "Related Definitions" at https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#efi-file-protocol-openex
+struct FileIOToken {
+    Event event;
+    Status status;
+    FlatPtr buffer_size;
+    void* buffer;
+};
+
+// EFI_FILE_PROTOCOL: https://uefi.org/specs/UEFI/2.10/13_Protocols_Media_Access.html#file-protocol
+struct FileProtocol {
+    using FileOpenFn = EFIAPI Status (*)(FileProtocol*, FileProtocol** new_handle, char16_t* file_name, FileOpenMode open_mode, FileAttribute attributes);
```

Comment:
No point in changing these occurrences since they are already written down, but are we expecting to use these types elsewhere?

Reply:
Assuming you mean the `using` declarations, no, we likely won't use these aliases anywhere else.
I just use them to keep the struct definitions more readable, as function pointer definitions are a IMHO bit weird to read. (And because the UEFI spec also uses function pointer aliases)

Reply:
Fair enough. :^)